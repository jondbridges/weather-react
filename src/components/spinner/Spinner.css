.Spinner-icon {
  padding: 0 10px;
}

.Spinner-text {
  font-size: 45px;
  font-weight: 100;
}
