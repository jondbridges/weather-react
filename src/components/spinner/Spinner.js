import React from 'react';
import FontAwesome from 'react-fontawesome';

const styles = {
  spinner: {
    padding: '0 10px'
  },
  spinnerText: {
    fontSize: '45px',
    fontWeight: 100
  }
};

const Spinner = ({displayText}) => {
  return (
    <div>
      <FontAwesome
        name='circle-o-notch'
        size='3x'
        spin
        style={styles.spinner}
      />
      <span style={styles.spinnerText}>{displayText}</span>
    </div>
  );
};

Spinner.propTypes = {
  displayText: React.PropTypes.string
};

export default Spinner;
