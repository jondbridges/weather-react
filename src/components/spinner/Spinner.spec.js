import React from 'react';
import Spinner from './Spinner';
import renderer from 'react-test-renderer';

describe('Spinner', () => {

  it('creates the spinner content', () => {
    const wrapper = renderer.create(
      <Spinner />
    );

    const tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('creates the spinner content with display text', () => {
    const wrapper = renderer.create(
      <Spinner displayText={"Loading..."} />
    );

    const tree = wrapper.toJSON();

    expect(tree).toMatchSnapshot();
  });
});