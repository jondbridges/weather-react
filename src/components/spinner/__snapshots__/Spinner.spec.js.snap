exports[`Spinner creates the spinner content 1`] = `
<div>
  <span
    className="fa fa-circle-o-notch fa-3x fa-spin Spinner-icon" />
  <span
    className="Spinner-text" />
</div>
`;

exports[`Spinner creates the spinner content with display text 1`] = `
<div>
  <span
    className="fa fa-circle-o-notch fa-3x fa-spin Spinner-icon" />
  <span
    className="Spinner-text">
    Loading...
  </span>
</div>
`;
